apps/build.groovy

```diff
@@ -134,10 +134,13 @@ EOF
 
     dir ('apps') {
         quietFlags = verboseMaven ? '' : '--quiet --batch-mode'
-        if (runTests) {
-            sh "mvn ${quietFlags} --threads 1C -Dmaven.javadoc.skip=true -Dcheckstyle.skip -Dmaven.build.cache.enabled=false -Dmaven.jacoco.skip=false clean install"
-        } else {
-            sh "mvn ${quietFlags} --threads 1C --update-snapshots -DskipITs -DskipTests -Dmaven.javadoc.skip=true -Dmaven.build.cache.enabled=false clean verify"
+        withEnv(['MAVEN_OPTS=-Xms1024m -Xmx1024m']) {
```

Comment:
This modest doubling of the heap size from 512m to 1024m for maven isn't unreasonable, but it's largely unrelated to the current issues of heap exhaustion. Instead, the resource exhaustion _seems_ to be occurring in the supporting infrastructure for testcontainers.org. This won't hurt, though!